File: Array/problems/easy/largest_3_distinct_element.go
package main

import (
	"fmt"
)

func print3Largest(arr []int) {
	size := len(arr)
	if size < 3 {
		fmt.Println("Invalid Array")
	}

	first := arr[0]
	second := arr[0]
	third := arr[0]
	for _, val := range arr {
		if first < val {
			third = second
			second = first
			first = val
		} else if second < val {
			third = second
			second = third
		} else if third < val {
			third = val
		}
	}

	fmt.Printf("%v %v %v\n", first, second, third)
}

func main() {
	arr := []int{12, 13, 1, 10, 34, 1}
	print3Largest(arr)
}

File: Array/problems/easy/largest_3_distinct_element.rs
fn main() {
    let arr = [12, 13, 1, 10, 34, 1];
    print_3_largest(&arr);
}

fn print_3_largest(arr: &[i32]) {
    let size = arr.len();
    if size < 3 {
        println!("Invalid Array");
        return;
    }
    let mut first = arr[0];
    let mut second = arr[0];
    let mut third = arr[0];
    for (_index, &val) in arr.iter().enumerate() {
        if first < val {
            third = second;
            second = first;
            first = val;
        } else if second < val {
            third = second;
            second = val;
        } else if third < val {
            third = val;
        }
    }

    println!("{} {} {}", first, second, third);
}
